pyjobshop/ProblemData.py

```diff
@@ -2,13 +2,66 @@
 from copy import deepcopy
 from dataclasses import dataclass, field, fields
 from itertools import pairwise
-from typing import Sequence, TypeVar
+from typing import Protocol, Sequence, TypeAlias, TypeVar
 
 from pyjobshop.constants import MAX_VALUE
+from pyjobshop.protocols import DataclassInstance, SizedDataclassInstance
 
 _T = TypeVar("_T")
 
+MISSING: Sequence = ()
```

Comment:
`MISSING` is misleading because its association with [`dataclass.MISSING`](https://docs.python.org/3/library/dataclasses.html#dataclasses.MISSING), which represents a missing default (i.e., a field that has no default).

It's clearer to explicitly use `()` in the defaults.